src/Api/Controllers/AuthController.php

```diff
@@ -142,4 +145,32 @@ public function changeUserEmail(string $hash): Response
 
         return $this->response($userData);
     }
+
+    /**
+     * Login user using Access Token
+     * @return Response
+     */
+    public function loginByAccessToken(): Response
+    {
+        $accessToken = $this->request->getPost('access_token','string');
+        $providerName =  $this->request->getPost('provider','string');
+
+        $source = Sources::findFirst([
+            'conditions'=>'title = ?0 and is_deleted = 0',
+            'bind'=>[$providerName]
+        ]);
+
+        if (!is_object($source)) {
```

Comment:
FindFirst method returns false if no record was found.
it is faster using and cleaner using !$source instead.